### PR TITLE
Allow model attributes to be accessed magically

### PR DESCRIPTION
**Pull request by rickywiens**
- I was running into a problem using Eloquent helper functions on these
models (pluck, filter, etc) and doing this will allow those functions to
work properly.
- General cleanup and formatting on all model files while I was in
there.
- Fixed a missing return statement
- Added a test for my trait.
- Removed scalar type hints to maintain php 5.6 compatibility.

### CODE REVIEW

File: src/Traits/HasAttributesTrait.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * @package BoldApps\ShopifyToolkit\Traits
+ */
+trait HasAttributesTrait
+{
+
+    public function __get($name)
+    {
+        if ($this->hasGetMutator($name)) {
+            return $this->{'get'.Str::studly($name)}();
+        }
```
Comment (James300) [inline src/Traits/HasAttributesTrait.php]: How will this be handled if it is false?

Comment (James300) [reply src/Traits/HasAttributesTrait.php]: Nevermind the tests show that it returns null, should it throw an exception or something more obvious?

File: src/Traits/HasAttributesTrait.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
```
Comment (walrusk) [inline src/Traits/HasAttributesTrait.php]: should this method maybe still return something if `hasGetMutator` is false?

Comment (James300) [reply src/Traits/HasAttributesTrait.php]: :spam:

File: src/Models/Order.php [hunk 3]
```diff
@@ -567,7 +574,7 @@ public function getDiscountCodes()
     }
 
     /**
-     * @return array
+     * @return object
```
Comment (walrusk) [inline src/Models/Order.php]: I think this should stay an array: https://help.shopify.com/api/reference/orders/order#note-attributes-property

Comment (rickywiens) [reply src/Models/Order.php]: yep

File: src/Models/CustomerAddress.php [hunk 4]
```diff
@@ -3,9 +3,13 @@
 namespace BoldApps\ShopifyToolkit\Models;
 
 use BoldApps\ShopifyToolkit\Contracts\Serializeable;
+use BoldApps\ShopifyToolkit\Traits\HasAttributesTrait;
 
 class CustomerAddress implements Serializeable
 {
+
```
Comment (bmansell) [inline src/Models/CustomerAddress.php]: Nitpick: extra newline.

File: src/Traits/HasAttributesTrait.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+
+    public function hasGetMutator($name)
+    {
+        return method_exists($this, 'get'.Str::studly($name));
+    }
+
+}
```
Comment (James300) [inline src/Traits/HasAttributesTrait.php]: No newline here

File: tests/Traits/HasAttributesTraitTest.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        $this->assertTrue($this->customerObject->hasGetMutator('firstName'));
+    }
+
+    public function testHasMutatorFails()
+    {
+        $this->assertFalse($this->customerObject->hasGetMutator('yesnt_exists'));
+        $this->assertFalse($this->customerObject->hasGetMutator('yesntExists'));
+    }
+
+}
```
Comment (James300) [inline tests/Traits/HasAttributesTraitTest.php]: Same, no newline

File: src/Models/CustomerAddress.php (hunk 4 above)
Comment (rickywiens) [reply src/Models/CustomerAddress.php]: The codestyle rules allow for either right now, if you want to change it commit a php codesniffer config!

**Pull request merged**